components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/connector/UserClaimUpdateConfigImpl.java

```diff
@@ -0,0 +1,185 @@
+    public Map<String, String> getPropertyNameMapping() {
+
+        Map<String, String> nameMapping = new HashMap<>();
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                "Enable user email verification on update");
+        nameMapping.put(IdentityRecoveryConstants.ConnectorConfig.EMAIL_VERIFICATION_ON_UPDATE_EXPIRY_TIME,
+                "Email verification on update code expiry time");
+        return nameMapping;
+    }
+
+    @Override
+    public Map<String, String> getPropertyDescriptionMapping() {
+
+        Map<String, String> descriptionMapping = new HashMap<>();
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                "Trigger a verification notification when user's email address is updated.");
+        descriptionMapping.put(IdentityRecoveryConstants.ConnectorConfig.EMAIL_VERIFICATION_ON_UPDATE_EXPIRY_TIME,
+                "Set the number of minutes the email verification mail would be valid. (Negative value for infinite " +
+                        "validity)");
+        return descriptionMapping;
+    }
+
+    @Override
+    public String[] getPropertyNames() {
+
+        List<String> properties = new ArrayList<>();
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE);
+        properties.add(IdentityRecoveryConstants.ConnectorConfig.EMAIL_VERIFICATION_ON_UPDATE_EXPIRY_TIME);
+        return properties.toArray(new String[0]);
+    }
+
+    @Override
+    public Properties getDefaultPropertyValues(String tenantDomain) {
+
+        String enableEmailVerificationOnUpdate = DEFAULT_ENABLE_VALUE_FOR_EMAIL_VERIFICATION_ON_UPDATE;
+        String emailVerificationOnUpdateCodeExpiry = DEFAULT_EMAIL_VERIFICATION_ON_UPDATE_CODE_EXPIRY_TIME;
+
+        readConfigurationValuesFromFile();
+
+        if (StringUtils.isNotBlank(enableEmailVerificationOnUpdateProperty)) {
+            enableEmailVerificationOnUpdate = enableEmailVerificationOnUpdateProperty;
+        }
+        if (StringUtils.isNotBlank(emailVerificationOnUpdateCodeExpiryProperty)) {
+            emailVerificationOnUpdateCodeExpiry = emailVerificationOnUpdateCodeExpiryProperty;
+        }
+
+        Properties properties = new Properties();
+        properties.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                enableEmailVerificationOnUpdate);
+        properties.put(IdentityRecoveryConstants.ConnectorConfig.EMAIL_VERIFICATION_ON_UPDATE_EXPIRY_TIME,
+                emailVerificationOnUpdateCodeExpiry);
+        return properties;
+    }
+
+    @Override
+    public Map<String, String> getDefaultPropertyValues(String[] propertyNames, String tenantDomain) throws
+            IdentityGovernanceException {
+
+        Properties properties = getDefaultPropertyValues(tenantDomain);
+        Map<String, String> defaultProperties = new HashMap<>();
```

Comment:
Shouldn't this rather be filteredDefaultProperties ?

Reply:
changed the name.